includes/class-wc-admin-notes-historical-data.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * WooCommerce Admin: Historical Analytics Data Note.
+ *
+ * Adds a notes to store alerts area concerning the historial analytics data tool.
+ *
+ * @package WooCommerce Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Historical_Data.
+ */
+class WC_Admin_Notes_Historical_Data {
+	const NOTE_NAME  = 'wc-admin-historical-data';
+
+	/**
+	 * Creates a note for regenerating historical data.
+	 */
+	public static function add_note() {
+		$data_store = WC_Data_Store::load( 'admin-note' );
+
+		// First, see if we've already created this kind of note so we don't do it again.
+		$note_ids = $data_store->get_notes_with_name( self::NOTE_NAME );
+		foreach ( (array) $note_ids as $note_id ) {
```

Comment:
Could we just check to see if `$note_ids` is not empty here instead of looping? If there are any notes with the name we could assume it has already been shown right?

Reply:
Yeah good call - this was from an example note, but we really only need to check if it exists. I cleaned this up along with `content_data`.